src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,92 @@
 package core.basesyntax;
 
+import java.util.Objects;
+
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final float DEFAULT_LOAD_FACTOR = 0.75f;
+    private Node<K, V>[] table;
+    private int capacity = 16;
+    private int threshold;
+    private int size;
+
+    public MyHashMap() {
+        table = new Node[capacity];
+
+    }
 
     @Override
     public void put(K key, V value) {
-
+        Node<K, V> node = new Node<>(key, value, null);
+        int index = hash(node.key);
+        isNeedToResize();
+        if (table[index] == null) {
+            table[index] = node;
+            size++;
+        } else {
+            Node<K, V> newNode = table[index];
+            while (newNode != null) {
+                if (Objects.equals(node.key, newNode.key)) {
+                    newNode.value = value;
+                    return;
+                }
+                if (newNode.next == null) {
+                    newNode.next = node;
+                    size++;
+                    return;
+                }
+                newNode = newNode.next;
+            }
+        }
     }
 
     @Override
     public V getValue(K key) {
+        for (Node<K, V> node : table) {
+            while (node != null) {
+                if (Objects.equals(node.key, key)) {
+                    return node.value;
+                }
+                node = node.next;
+            }
+        }
+
         return null;
     }
 
     @Override
     public int getSize() {
-        return 0;
+        return size;
+    }
+
+    private int hash(K key) {
```

Comment:
actually this method return an index instead of hash so let's rename it again :)